src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -263,7 +263,9 @@ function assertValidProps(component, props) {
   }
   invariant(
     props.style == null || typeof props.style === 'object',
-    'The `style` prop expects a mapping from style properties to values, ' +
+    'Please check the render method of ' +
+     component._currentElement._owner._instance.__proto__.constructor.displayName +
+    '. The `style` prop expects a mapping from style properties to values, ' +
     'not a string. For example, style={{marginRight: spacing + \'em\'}} when ' +
     'using JSX.'
```

Comment:
Can you just use `getDeclarationErrorAddendum` here. That will wrap up the `_owner` stuff and get the name.